impl/src/main/java/org/jboss/weld/bootstrap/ContextHolder.java

```diff
@@ -14,7 +18,8 @@
     public ContextHolder(T context, Class<T> type, Set<Annotation> qualifiers) {
         this.context = context;
         this.type = type;
-        this.qualifiers = qualifiers;
+        this.qualifiers = asSet(DefaultLiteral.INSTANCE, AnyLiteral.INSTANCE);
```

Comment:
This is likely unnecessary. The set of qualifiers is expected to contain all elements. See also `org.jboss.weld.bootstrap.WeldStartup.createContexts(ServiceRegistry)` and other places where this constructor is used.

Reply:
You are likely right, I just meant to be rather safe then sorry, I will remove it and we'll see how it goes.